checksum/checksum.go

```diff
@@ -123,13 +123,13 @@ func (c Checksum) WriteToV2(m *refs.Checksum) {
 // See also [NewTillichZemor], [NewSHA256].
 func (c Checksum) Type() Type {
 	v2 := (refs.Checksum)(c)
-	switch v2.GetType() {
+	switch typ := v2.GetType(); typ {
 	case refs.SHA256:
 		return SHA256
 	case refs.TillichZemor:
 		return TZ
 	default:
-		return Unknown
+		return Type(typ)
```

Comment:
what is motivation? cant it lead to bugs? some code now may operate sums it does not know

Reply:
> what is motivation?

as written in the commit - to not lose the data and work with that we got. Nothing prevents to get real value, not zerofy it

> cant it lead to bugs? some code now may operate sums it does not know

if code is written correctly - no bugs. If incorrectly - the code is already buggy. Good point to detect potential problems